sdk/tables/azure-tables/src/TableClient.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TableServiceClient } from "./TableServiceClient";
+import {
+  Entity,
+  ListEntitiesOptions,
+  CreateEntityOptions,
+  UpdateEntityOptions,
+  MergeEntityOptions,
+  SetAccessPolicyOptions
+} from "./models";
+import {
+  TableServiceClientOptions,
+  QueryOptions,
+  GetEntityOptions,
+  GetEntityResponse,
+  ListEntitiesResponse,
+  CreateEntityResponse,
+  DeleteEntityOptions,
+  DeleteEntityResponse,
+  UpdateEntityResponse,
+  MergeEntityResponse,
+  GetAccessPolicyOptions,
+  GetAccessPolicyResponse,
+  SignedIdentifier,
+  SetAccessPolicyResponse
+} from "./generatedModels";
+
+/**
+ * A TableClient represents a Client to the Azure Tables service allowing you
+ * to perform operations on a single table.
+ */
+export class TableClient {
+  private client: TableServiceClient;
+  private tableName: string;
```

Comment:
thoughts on making the `tableName` property `public readonly`?

Reply:
Yeah, I think that would be a good idea